sympy/functions/elementary/integers.py

```diff
@@ -517,6 +517,53 @@ def _eval_is_zero(self):
     def _eval_is_negative(self):
         return False
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.calculus.accumulationbounds import AccumBounds
+        arg = self.args[0]
+        arg0 = arg.subs(x, 0)
+        r = self.subs(x, 0)
+
+        if arg0.is_finite:
+            if arg0.is_integer or r.is_zero:
```

Comment:
Is there some case where `arg0.is_integer` does not return `True` but `r.is_zero` returns `True`?